### ISSUE TITLE
Froze atom on repo with many files (thanks to node_module) because atom-hg is trying to run hg status -A

### ISSUE THREAD

**Issue by v3ss0n** [victor-torres/atom-hg#27]
https://github.com/victor-torres/atom-hg/blob/28f84975590425094def66377e92e0163649d528/lib/hg-utils.coffee#L299
causes every files to check for versioning info , including node_module (even tho ignored) which cause atom froze .
I think it should run in a separate process ?

**Comment by TomKemperNL** [comment]
Yep, I'm working on that. Do you have any particular open source node-repo that demonstrates this especially well?

I'm testing on a few electron repos but the freezes are only 250-400ms for me, which is annoying but kinda hard to measure improvements with.

**Comment by v3ss0n** [comment]
for me , when re-opening atom , it toally froze  ,. i waited for about 1 hr. it is emberjs project with about 10 ember addon , whcih have recursive dependencies inside nodemodules thanks to brocolli.

**Comment by TomKemperNL** [comment]
That's horrible!

I'm currently evaluating different options on improving performance. I'm unfamiliar with Ember unfortunately, do you know of an OSS repo I could clone to reproduce this?

I've just merged a full asynching of hg stat -A to my own fork (which I would generally consider 'unstable').

You're welcome to see if it helps though:
apm uninstall atom-hg
apm install tomkempernl/atom-hg

Unfortunately, I would expect those changes to improve seconds... not the kinds of troubles you're describing.

**Comment by victor-torres** [comment]
So the dependencies are not versionsed (as they shouldn't be) but it's taking too long to load because of them... Maybe as a workaround we could manually (with code) ignore these directories.

**Comment by v3ss0n** [comment]
@victor-torres  yes they aren't versioned.

**Comment by v3ss0n** [comment]
i think respecting .hgignore will be good enough , acutally hg status -A should does that , i gonna open a but on mercurial repo.

**Comment by victor-torres** [comment]
@v3ss0n have you tried the newest version? May I close this issue?

**Comment by Goodwine** [comment]
In my case, the problem comes from `hg cat`, should I create a separate issue for that?

**Comment by v3ss0n** [comment]
i haven't update yet , i will try. 
Actually i had disable it just so i can still open atom :smile:

**Comment by victor-torres** [comment]
Any update about this issue, @v3ss0n?
